aws/ami/build_ami.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash -e
+
+PRODUCT=scylla
+DIR=$(dirname $(readlink -f $0))
+
+print_usage() {
+    echo "build_ami.sh --localrpm --repo [URL] --target [distribution]"
+    echo "  --localrpm  deploy locally built rpms"
+    echo "  --repo  repository for both install and update, specify .repo/.list file URL"
+    echo "  --repo-for-install  repository for install, specify .repo/.list file URL"
+    echo "  --repo-for-update  repository for update, specify .repo/.list file URL"
+    echo "  --product          scylla or scylla-enterprise"
+    exit 1
+}
+LOCALRPM=0
+REPO_FOR_INSTALL=
+while [ $# -gt 0 ]; do
+    case "$1" in
+        "--localrpm")
+            LOCALRPM=1
+            shift 1
+            ;;
+        "--repo")
+            REPO_FOR_INSTALL=$2
+            INSTALL_ARGS="$INSTALL_ARGS --repo $2"
+            shift 2
+            ;;
+        "--repo-for-install")
+            REPO_FOR_INSTALL=$2
+            INSTALL_ARGS="$INSTALL_ARGS --repo-for-install $2"
+            shift 2
+            ;;
+        "--repo-for-update")
+            INSTALL_ARGS="$INSTALL_ARGS --repo-for-update $2"
+            shift 2
+            ;;
+        "--product")
+            PRODUCT=$2
+            shift 2
+            ;;
+        *)
+            print_usage
+            ;;
+    esac
+done
+
+is_redhat_variant() {
+    [ -f /etc/redhat-release ]
+}
+is_debian_variant() {
+    [ -f /etc/debian_version ]
+}
+
+pkg_install() {
+    if is_redhat_variant; then
+        sudo yum install -y $1
+    elif is_debian_variant; then
+        sudo apt-get install -y $1
+    else
+        echo "Requires to install following command: $1"
+        exit 1
+    fi
+}
+
+get_version_from_rpm () {
+    RPM=$1
+    RELEASE=$(rpm -qi $RPM | awk '/Release/ { print $3 }' )
+    VERSION=$(rpm -qi $RPM | awk '/Version/ { print $3 }' )
+    echo "$VERSION-$RELEASE"
+}
+AMI=ami-ae7bfdb8
+REGION=us-east-1
+SSH_USERNAME=centos
+
+if [ $LOCALRPM -eq 1 ]; then
+    # TODO move to new repo
+    # REPO=`./scripts/scylla_current_repo --target centos`
+    INSTALL_ARGS="$INSTALL_ARGS --localrpm"
+
+    SCYLLA_VERSION=$(get_version_from_rpm $DIR/files/$PRODUCT-server*.x86_64.rpm)
```

Comment:
We need to check *.rpm file existence.
When I tried ./aws/ami/build_ami.sh --localrpm, I mistakenly copied *.rpm to another directory, the script just proceed to build AMI and failed.

Reply:
Will add a check

Reply:
added a check